File: app/src/main/java/com/example/farmer/fertilizer/FertilizerExpenditure.java
package com.example.farmer.fertilizer;


import org.json.JSONException;
import org.json.JSONObject;

public class FertilizerExpenditure {
    private String itemName;
    private String purchaseDate;
    private String purchaseAmount; // Field for the purchase amount
    private String paymentMode;
    private String receiptImagePath;

    // Constructor
    public FertilizerExpenditure(String itemName, String purchaseDate, String purchaseAmount, String paymentMode, String receiptImagePath) {
        this.itemName = itemName;
        this.purchaseDate = purchaseDate;
        this.purchaseAmount = purchaseAmount;
        this.paymentMode = paymentMode;
        this.receiptImagePath = receiptImagePath;
    }

    // Getter methods
    public String getItemName() {
        return itemName;
    }

    public String getPurchaseDate() {
        return purchaseDate;
    }

    public String getPurchaseAmount() { // Getter for purchase amount
        return purchaseAmount;
    }

    public String getAmount()
    {
        return purchaseAmount;
    }

    public String getPaymentMode() {
        return paymentMode;
    }

    public String getReceiptImagePath() {
        return receiptImagePath;
    }

    // Static method to create an instance from a JSON object
    public static FertilizerExpenditure fromJson(JSONObject jsonObject) {
        try {
            String itemName = jsonObject.optString("ItemName", "");
            String purchaseDate = jsonObject.optString("PurchaseDate", "");
            String purchaseAmount = jsonObject.optString("PurchaseAmount", "");
            String paymentMode = jsonObject.optString("PaymentMode", "");
            String receiptPath = jsonObject.optString("ReceiptPath", ""); // Use "ReceiptPath" to match saveData()

            return new FertilizerExpenditure(itemName, purchaseDate, purchaseAmount, paymentMode, receiptPath);
        } catch (Exception e) {
            e.printStackTrace();
            return null;
        }
    }
}

File: app/src/main/java/com/example/farmer/fertilizer/FertilizerExpenditure.kt
package com.example.farmer.fertilizer

import org.json.JSONObject


class FertilizerExpenditure // Constructor
    (
// Getter methods
    @JvmField val itemName: String?, @JvmField val purchaseDate: String?, // Getter for purchase amount
    val amount: String?, // Field for the purchase amount
    @JvmField val paymentMode: String?, @JvmField val receiptImagePath: String?
) {
    companion object {
        // Static method to create an instance from a JSON object
        @JvmStatic
        fun fromJson(jsonObject: JSONObject): FertilizerExpenditure? {
            try {
                val itemName = jsonObject.optString("ItemName", "")
                val purchaseDate = jsonObject.optString("PurchaseDate", "")
                val purchaseAmount = jsonObject.optString("PurchaseAmount", "")
                val paymentMode = jsonObject.optString("PaymentMode", "")
                val receiptPath =
                    jsonObject.optString("ReceiptPath", "") // Use "ReceiptPath" to match saveData()

                return FertilizerExpenditure(
                    itemName,
                    purchaseDate,
                    purchaseAmount,
                    paymentMode,
                    receiptPath
                )
            } catch (e: Exception) {
                e.printStackTrace()
                return null
            }
        }
    }
}
